com.minres.coredsl/src/com/minres/coredsl/scoping/CoreDslScopeProvider.xtend

```diff
@@ -0,0 +1,258 @@
+package com.minres.coredsl.scoping
+
+import com.minres.coredsl.coreDsl.BitField
+import com.minres.coredsl.coreDsl.CompoundStatement
+import com.minres.coredsl.coreDsl.CoreDef
+import com.minres.coredsl.coreDsl.CoreDslPackage
+import com.minres.coredsl.coreDsl.Declaration
+import com.minres.coredsl.coreDsl.DeclarationStatement
+import com.minres.coredsl.coreDsl.Declarator
+import com.minres.coredsl.coreDsl.DescriptionContent
+import com.minres.coredsl.coreDsl.EntityReference
+import com.minres.coredsl.coreDsl.EnumTypeDeclaration
+import com.minres.coredsl.coreDsl.Expression
+import com.minres.coredsl.coreDsl.ForLoop
+import com.minres.coredsl.coreDsl.FunctionCallExpression
+import com.minres.coredsl.coreDsl.FunctionDefinition
+import com.minres.coredsl.coreDsl.ISA
+import com.minres.coredsl.coreDsl.Instruction
+import com.minres.coredsl.coreDsl.InstructionSet
+import com.minres.coredsl.coreDsl.IntrinsicExpression
+import com.minres.coredsl.coreDsl.MemberAccessExpression
+import com.minres.coredsl.coreDsl.NamedEntity
+import com.minres.coredsl.coreDsl.ParenthesisExpression
+import com.minres.coredsl.coreDsl.StructTypeDeclaration
+import com.minres.coredsl.coreDsl.StructTypeSpecifier
+import com.minres.coredsl.coreDsl.TypeSpecifier
+import com.minres.coredsl.coreDsl.UnionTypeDeclaration
+import com.minres.coredsl.coreDsl.UnionTypeSpecifier
+import com.minres.coredsl.coreDsl.UserTypeDeclaration
+import java.util.HashSet
+import org.eclipse.emf.ecore.EClass
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.emf.ecore.EReference
+import org.eclipse.xtext.scoping.IScope
+import org.eclipse.xtext.scoping.Scopes
+
+import static extension com.minres.coredsl.util.ModelExtensions.*
+
+class CoreDslScopeProvider extends AbstractCoreDslScopeProvider {
+// Possible references:
+//   InstructionSet.superType
+//   CoreDef.providedInstructionSets
+//   StructTypeSpecifier.target
+//   UnionTypeSpecifier.target
+//   EnumTypeSpecifier.target
+//   MemberAccessExpression.declarator
+//   EntityReference.target
+	override IScope getScope(EObject context, EReference reference) {
+
+		println('''Trying to resolve «reference.EType.name» «(reference.eContainer as EClass).name».«reference.name» on «context.eClass.name»''');
```

Comment:
Doesn't this print a lot of output? Maybe wrap this in the Xtext-equivalent of an `#ifdef DEBUG`.

Reply:
Yeah I forgot to remove that.